File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/entity/Organization.java
package pers.acp.admin.oauth.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.hibernate.annotations.GenericGenerator;
import pers.acp.admin.oauth.base.OauthBaseTreeEntity;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

/**
 * @author zhangbin by 2018-1-17 16:39
 * @since JDK 11
 */
@Entity
@Table(name = "t_organization")
@ApiModel("机构信息")
public class Organization extends OauthBaseTreeEntity<Organization> {

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public int getSort() {
        return sort;
    }

    public void setSort(int sort) {
        this.sort = sort;
    }

    public Set<User> getUserSet() {
        return userSet;
    }

    public void setUserSet(Set<User> userSet) {
        this.userSet = userSet;
    }

    public Set<User> getAdminUserSet() {
        return adminUserSet;
    }

    public void setAdminUserSet(Set<User> adminUserSet) {
        this.adminUserSet = adminUserSet;
    }

    @Id
    @GenericGenerator(name = "idGenerator", strategy = "uuid")
    @GeneratedValue(generator = "idGenerator")
    @Column(length = 36, nullable = false)
    @ApiModelProperty("机构ID")
    private String id;

    @Column(nullable = false)
    @ApiModelProperty("机构名称")
    private String name;

    @Column(length = 100, nullable = false)
    @ApiModelProperty("机构编码")
    private String code = "";

    @Column(nullable = false)
    @ApiModelProperty("序号")
    private int sort;

    @JsonIgnore
    @ManyToMany(fetch = FetchType.LAZY, cascade = {CascadeType.REFRESH})
    @JoinTable(name = "t_user_organization_set",
            joinColumns = {@JoinColumn(name = "organizationid", referencedColumnName = "id")},
            inverseJoinColumns = {@JoinColumn(name = "userid", referencedColumnName = "id")})
    private Set<User> userSet = new HashSet<>();

    @JsonIgnore
    @ManyToMany(fetch = FetchType.LAZY, cascade = {CascadeType.REFRESH})
    @JoinTable(name = "t_user_organization_mng_set",
            joinColumns = {@JoinColumn(name = "organizationid", referencedColumnName = "id")},
            inverseJoinColumns = {@JoinColumn(name = "userid", referencedColumnName = "id")})
    private Set<User> adminUserSet = new HashSet<>();

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/entity/Organization.kt
package pers.acp.admin.oauth.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.hibernate.annotations.GenericGenerator
import pers.acp.admin.oauth.base.OauthBaseTreeEntity

import javax.persistence.*

/**
 * @author zhangbin by 2018-1-17 16:39
 * @since JDK 11
 */
@Entity
@Table(name = "t_organization", indexes = [
    Index(columnList = "area"),
    Index(columnList = "code"),
    Index(columnList = "parentId")])
@ApiModel("机构信息")
data class Organization(
        @Id
        @GenericGenerator(name = "idGenerator", strategy = "uuid")
        @GeneratedValue(generator = "idGenerator")
        @Column(length = 36, nullable = false)
        @ApiModelProperty("机构ID")
        var id: String = "",

        @Column(nullable = false)
        @ApiModelProperty("机构名称")
        var name: String = "",

        @Column(length = 100, nullable = false)
        @ApiModelProperty("机构区域")
        var area: String = "",

        @Column(length = 100, nullable = false)
        @ApiModelProperty("机构编码")
        var code: String = "",

        @Column(nullable = false)
        @ApiModelProperty("序号")
        var sort: Int = 0,

        @JsonIgnore
        @ManyToMany(fetch = FetchType.LAZY, cascade = [CascadeType.REFRESH])
        @JoinTable(name = "t_user_organization_set", joinColumns = [JoinColumn(name = "organizationId", referencedColumnName = "id")], inverseJoinColumns = [JoinColumn(name = "userId", referencedColumnName = "id")])
        var userSet: MutableSet<User> = mutableSetOf(),

        @JsonIgnore
        @ManyToMany(fetch = FetchType.LAZY, cascade = [CascadeType.REFRESH])
        @JoinTable(name = "t_user_organization_mng_set", joinColumns = [JoinColumn(name = "organizationId", referencedColumnName = "id")], inverseJoinColumns = [JoinColumn(name = "userId", referencedColumnName = "id")])
        var adminUserSet: MutableSet<User> = mutableSetOf()
) : OauthBaseTreeEntity<Organization>() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val organization = other as Organization
        return EqualsBuilder()
                .appendSuper(super.equals(other))
                .append(id, organization.id)
                .append(name, organization.name)
                .append(area, organization.area)
                .append(code, organization.code)
                .append(sort, organization.sort)
                .append(parentId, organization.parentId)
                .isEquals
    }

    override fun hashCode(): Int =
            HashCodeBuilder(17, 37)
                    .appendSuper(super.hashCode())
                    .append(id)
                    .append(name)
                    .append(area)
                    .append(code)
                    .append(sort)
                    .append(parentId)
                    .toHashCode()

    override fun toString(): String =
            StringBuilder("Organization(")
                    .append("id=$id")
                    .append(",name=$name")
                    .append(",area=$area")
                    .append(",code=$code")
                    .append(",sort=$sort")
                    .append(",parentId=$parentId")
                    .append(")")
                    .toString()
}